models/commons/src/reference/types.ts

```diff
@@ -0,0 +1,22 @@
+import type { TBuilder } from '@commercetools-test-data/core';
+
+export type TReferenceBuilder = TBuilder<TReference>;
+
+export type TCreateReferenceBuilder = () => TReferenceBuilder;
+
+type TExpandedReferenceObject = {
+  id: string;
+  [key: string]: unknown;
+};
+
+export type TReference = {
+  typeId: string;
+  id: TExpandedReferenceObject['id'];
+  obj?: TExpandedReferenceObject;
```

Comment:
It would be better to pass a generic here. If it's too complicated, we can leave it like that, or just type it as `obj?: unknown`

Reply:
could it be something like this?

```
export type TReference<T extends TExpandedReferenceObject> = {
  typeId: string;
  id: T['id'];
  obj?: T;
};
```

Reply:
Something like that. The `id` can simply be typed as a `string`

Reply:
https://github.com/commercetools/test-data/pull/58/commits/2986fa323bf13fb7ac0280aa3025c101cdee9d46